catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/cache/solr/impl/CachingFederationStrategyTest.java

```diff
@@ -0,0 +1,390 @@
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import org.codice.ddf.itests.common.WaitCondition;
+import org.geotools.filter.NullFilterImpl;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.opengis.filter.sort.SortBy;
+
+import ddf.catalog.Constants;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.ResultImpl;
+import ddf.catalog.filter.proxy.builder.GeotoolsFilterBuilder;
+import ddf.catalog.impl.MockSource;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.DeleteResponseImpl;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.QueryResponseImpl;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+import ddf.catalog.operation.impl.UpdateResponseImpl;
+import ddf.catalog.plugin.PreFederatedQueryPlugin;
+import ddf.catalog.source.Source;
+
+public class CachingFederationStrategyTest {
+
+    private static final long LONG_TIMEOUT = TimeUnit.MINUTES.toMillis(1);
+
+    private ExecutorService executor, queryExecutor;
+
+    private Query mockQuery;
+
+    private CachingFederationStrategy strategy;
+
+    private PreFederatedQueryPlugin preQueryPlugin;
+
+    @Before
+    public void setup() throws Exception {
+        executor = Executors.newSingleThreadExecutor();
+        queryExecutor = Executors.newSingleThreadExecutor();
+
+        preQueryPlugin = mock(PreFederatedQueryPlugin.class);
+        when(preQueryPlugin.process(any(),
+                any())).thenAnswer(invocation -> invocation.getArguments()[1]);
+
+        strategy = new CachingFederationStrategy(queryExecutor,
+                Arrays.asList(preQueryPlugin),
+                new ArrayList<>(),
+                new SolrCache(mock(SolrClientAdaptor.class)),
+                executor,
+                mock(ValidationQueryFactory.class),
+                new CacheQueryFactory(new GeotoolsFilterBuilder())) {
+            protected Runnable createMonitor(
+                    final CompletionService<SourceResponse> completionService,
+                    final Map<Future<SourceResponse>, QueryRequest> futures,
+                    final QueryResponseImpl returnResults, final QueryRequest request) {
+
+                return new SortedQueryMonitor(this,
+                        completionService,
+                        futures,
+                        returnResults,
+                        request,
+                        postQuery) {
+                    @Override
+                    public void run() {
+                        returnResults.setHits(42);
```

Comment:
It is just an arbitrary value that is used later for assertions. @ahoffer and I decided to override the `run()` method in `SortedQueryMonitor` because, while a `SortedQueryMonitor` instance needs to be created and ran in `CachingFederationStrategy` for full coverage, the `CachingFederationStrategyTest` unit tests are not concerned with the behavior in `SortedQueryMonitor`. Honestly, this line could be deleted, but then there would be nothing to `assert` on. It is also helpful in simply showing that a `SortedQueryMonitor` is created and `run()`.